core/src/main/java/hudson/model/View.java

```diff
@@ -463,9 +463,18 @@ private boolean isRelevant(Collection<Label> labels, Computer computer) {
         Collection<TopLevelItem> items = getItems();
         List<Queue.Item> result = new ArrayList<Queue.Item>();
         for (Queue.Item qi : base) {
-            if (items.contains(qi.task)) {
-                result.add(qi);
-            } else
```

Comment:
Deliberately getting rid of the `else` and thus running the `if (qi.task instanceof AbstractProject<?, ?>)` block even if we added to `result` in the loop here? That seems suspicious. If I read this correctly, for a typical `FreestyleProject` you are going to add two entries.

Maybe you meant to add `SOMELABEL:` to the `for`-loop and `continue` it, rather than `break`ing out of the `do` loop and falling through to the `AbstractProject` check.

Reply:
You are correct, good catch! A label crossed my mind but didn't know it was kosher to use in Jenkins code base! I will take this and your other comment into consideration and try to redesign the loop. Thanks!